File: C++/Czesciowe ulatwienie.cpp
#include <iostream>

int main() 
{
    int Z, n, count;
    std::cin >> Z;
    while (Z--) 
    {
        std::cin >> n;
        int nimSum = 0;
        for (int i = 0; i < n; i++) 
        {
            std::cin >> count;
            nimSum ^= count;
        }
        if (nimSum == 0)
            std::cout << 2 << '\n'; 
        else
           std::cout << 1 << '\n';
    }
    return 0;
}

File: Python/Czesciowe ulatwienie.py
z = int(input())
for _ in range(z):
    n = int(input())
    line_counts = list(map(int, input().split()))
    nim_sum = 0
    for count in line_counts:
        nim_sum ^= count
    if nim_sum == 0:
        print(2)
    else:
        print(1)